imagepuller/internal/auth/auth.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2025 Edgeless Systems GmbH
+// SPDX-License-Identifier: BUSL-1.1
+
+package auth
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/BurntSushi/toml"
+	"github.com/edgelesssys/contrast/imagepuller/internal/api"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+)
+
+// Config represents the imagepuller's registry authentication configurations.
+type Config struct {
+	Registries map[string]Registry
+	ExtraEnv   map[string]string
+}
+
+// Registry represents authentication configuration for a single registry.
+type Registry struct {
+	authn.AuthConfig
+	CACerts            string
+	InsecureSkipVerify bool
+}
```

Comment:
Please add some toml marshalling hints (I believe the convention is `kebab-case`) and verify that the JSON marshalling hints for [`AuthConfig`](https://pkg.go.dev/github.com/google/go-containerregistry/pkg/authn#AuthConfig) are respected when unmarshalling TOML.